PSReadLine/Accessibility.cs

```diff
@@ -10,14 +11,73 @@ internal class Accessibility
     {
         internal static bool IsScreenReaderActive()
         {
-            bool returnValue = false;
-
             if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                return IsAnyWindowsScreenReaderEnabled();           // Check for macOS VoiceOver
+
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+                return IsVoiceOverEnabled();
+
+            // TODO: Support Linux per https://code.visualstudio.com/docs/configure/accessibility/accessibility
+            return false;
+        }
+
+        private static bool IsAnyWindowsScreenReaderEnabled() {
+            // The official way to check for a screen reader on Windows is
+            // SystemParametersInfo(SPI_GETSCREENREADER, ...) but it doesn't
+            // detect the in-box Windows Narrator and is otherwise known to be
+            // problematic.
+            //
+            // Unfortunately, the alternative method used by Electron and
+            // Chromium, where the relevant screen reader libraries (modules)
+            // are checked for does not work in the context of PowerShell
+            // because it relies on those applications injecting themselves into
+            // the app. Which they do not because it's not a windowed app, so
+            // we're stuck using the known-to-be-buggy way.
+            bool spiScreenReader = false;
+            PlatformWindows.SystemParametersInfo(PlatformWindows.SPI_GETSCREENREADER, 0, ref spiScreenReader, 0);
+            if (spiScreenReader)
+                return true;
+
+            // At least we can correctly check for Windows Narrator using the
+            // NarratorRunning mutex (which is mostly not broken with
+            // PSReadLine, as it were).

```

Comment:
> which is mostly not broken with PSReadLine, as it were

I'm not super sure what you mean by this :) Can you please reword it a bit?

Reply:
Windows Narrator is mostly not broken with PSReadLine. Not in the way that NVDA and VoiceOver are.

Reply:
It seems to be programmed to ignore duplicate output, the whole re-rendering issue doesn't present itself under Narrator. Which made figuring out "what's wrong" (given no reproduction steps) a tad difficult.

Reply:
Minor update based on the conversation.
```suggestion
            // NarratorRunning mutex. Windows Narrator is mostly not broken with
            // PSReadLine, not in the way that NVDA and VoiceOver are.

```